### ISSUE TITLE
vault fails on token - eks - terraform - jx boot

### ISSUE THREAD

**Issue by arpad9** [jenkins-x/jx#7510]
### Summary 

After creating an eks cluster using terraform and then running `jx boot`, the vault pod never comes up.

### Steps to reproduce the behavior

1. Create eks cluster using terraform.
2. set jx-requirements.yml with the output from terraform
3. run `jx boot`

stanza in jx-requirements.yml:
```
vault:
  aws:
    dynamoDBRegion: us-east-1
    dynamoDBTable: vault-unseal-jenkins-<redacted>
    iamUserName: <redacted>
    kmsKeyId: <redacted>
    kmsRegion: us-east-1
    s3Bucket: vault-unseal-jenkins-<redacted>
    s3Region: us-east-1
```
The following is also set:
```
VAULT_SEAL_TYPE=awskms
VAULT_AWSKMS_SEAL_KEY_ID=<redacted>
VAULT_AWS_SECRET_ACCESS_KEY=<redacted>
VAULT_AWS_ACCESS_KEY_ID=<redacted>
```

### Expected behavior
Vault comes up.

### Actual behavior

`Waiting for vault to be initialized and unsealed...`

Further investigation reveals that the vault container isn't getting the correct token:
`kubectl -n jx logs -f  jx-vault-jenkins-0 vault`
```
Using eth0 for VAULT_CLUSTER_ADDR: https://10.0.1.92:8201
telemetry.disable_hostname has been set to false. Recommended setting is true for Prometheus to avoid poorly named metrics.
Error initializing storage of type dynamodb: UnrecognizedClientException: The security token included in the request is invalid.
        status code: 400, request id: <redacted>
```

### Jx version

The output of `jx version` is:

```
Version        2.1.127
Commit         4bc05a9
Build date     2020-08-05T20:34:57Z
Go version     1.13.8
Git tree state clean
```

### Diagnostic information

The output of `jx diagnose version` is:

```
Running in namespace: jx
WARNING: Failed to find helm installs: running helm list --all --namespace jx: failed to run 'helm list --all --namespace jx' command in directory '', output: 'Error: could not find tiller'
Version        2.1.127
Commit         4bc05a9
Build date     2020-08-05T20:34:57Z
Go version     1.13.8
Git tree state clean
NAME                          VERSION
Kubernetes cluster            v1.16.8-eks-fd1ea7
kubectl (installed in JX_BIN) v1.16.8
git                           2.25.1
Operating System              Ubuntu 20.04.1 LTS

Please visit https://jenkins-x.io/faq/issues/ for any known issues.

Finished printing diagnostic information.

```

### Kubernetes cluster

eks

### Kubectl version
The output of `kubectl version --client` is:

```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.8", GitCommit:"ec6eb119b81be488b030e849b9e64fda4caaf33c", GitTreeState:"clean", BuildDate:"2020-03-12T21:00:06Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
```

### Operating system / Environment

Ubuntu 20.04 LTS

**Comment by arpad9** [comment]
I've tried this several different ways now and it's always the same result.  I've had terraform skip creation of the table, bucket and key, which caused `jx boot` to create a CF stack for vault... same result.

**Comment by arpad9** [comment]
Also: I don't know if this deserves another Issue or not, but the vault policies that attach to the IAM user I give it keep piling up with each install.

**Comment by arpad9** [comment]
So... I haven't completely solved this because now I'm getting complaints about an invalid cert but I have made it past this.

TLDR;
The issue was having the wrong AWS auth and not being able to generate a temporary token to unseal vault.

Problems:
1. Unknown: I went back to start with a fresh cluster using terraform.  To save myself any mix-ups, I put the AWS user and secret into my bashrc.  Well... I reversed the values.  So I don't know what was happening before this but I know that having your key as your secret and your secret as your key doesn't work.

2. Great.  Fresh cluster.  Fresh jx boot.  Got stuck on unsealing because DNS hadn't propagated fast enough.  Once DNS propagated, it got a little further.

3. NOW, the vault ingress is using the default cert and `jx boot` or something doesn't like that:
```
error: creating system vault URL client: wait for vault to be initialized and unsealed: reading vault health: Get https://vault-jx.domain.com/v1/sys/health?drsecondarycode=299&performancestandbycode=299&sealedcode=299&standbycode=299&uninitcode=299: x509: certificate is valid for ingress.local, not vault-jx.domain.com
error: failed to interpret pipeline file jenkins-x.yml: failed to run '/bin/sh -c jx step create values --name parameters' command in directory 'env', output: ''
```
I've updated the ingress to point to our cert but no luck so far.

Still, the problem is outside of the scope of this Issue and I'm closing this one because I got past it.
